### PR TITLE
server: speed up unit test

### PR DESCRIPTION
**Pull request by tiancaiamao**
run many test cases parallel
split databases in server package for parallel

depends on https://github.com/pingcap/check/pull/7

### CODE REVIEW

**Comment by ngaut** [comment]
Any compare results ?

**Comment by tiancaiamao** [comment]
this is an small improvement, after Parallel(), server package unit test time from 50s down to 20s ...
although the dominant cost is still in ddl & store/tikv 
https://github.com/pingcap/tidb/issues/2008

File: server/tidb_test.go [hunk 1]
```diff
@@ -44,7 +45,21 @@ func (ts *TidbTestSuite) SetUpSuite(c *C) {
 	c.Assert(err, IsNil)
 	ts.server = server
 	go ts.server.Run()
-	time.Sleep(time.Millisecond * 100)
+	waitUntilServerOnline()
+
+	// run this test here because parallel would affect the result of it
```
Comment (zimulala) [inline server/tidb_test.go]: s/run/Run and add .

File: server/server_test.go [hunk 2]
```diff
@@ -55,6 +55,34 @@ func runTests(c *C, dsn string, tests ...func(dbt *DBTest)) {
...
+	c.Assert(err, IsNil, Commentf("Error use database %s", dbName))
+
+	dbt := &DBTest{c, db}
+	for _, test := range tests {
+		test(dbt)
+		dbt.db.Exec("DROP TABLE IF EXISTS test")
+	}
+
+	_, err = db.Exec(dropDB)
+	c.Assert(err, IsNil, Commentf("Error create database %s", dbName))
```
Comment (zimulala) [inline server/server_test.go]: s/create/drop

File: _vendor/src/github.com/pingcap/check/check.go [hunk 3]
```diff
@@ -611,13 +612,23 @@ func (runner *suiteRunner) run() *Result {
...
+					case <-c.done:
+					case <-c.parallel:
+						delayedC = append(delayedC, c)
+					}
 					if c.status() == fixturePanickedSt {
 						runner.skipTests(missedSt, runner.tests[i+1:])
 						break
 					}
 				}
+				// wait those parallel tests finish.
```
Comment (zimulala) [inline _vendor/src/github.com/pingcap/check/check.go]: s/wait/Wait

**Comment by zimulala** [comment]
Rest LGTM

**Comment by tiancaiamao** [comment]
PTAL @shenli @coocood @hanfei1991 @zimulala

**Comment by zimulala** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @shenli @coocood @hanfei1991 @XuHuaiyu

**Comment by coocood** [comment]
LGTM

**Pull request merged**